cppcache/src/Log.cpp

```diff
@@ -804,6 +804,217 @@ void LogVarargs::finest(const char* fmt, ...) {
   va_end(argp);
 }
 
+LogLevel Log::logLevel() { return s_logLevel; }
+
+void Log::setLogLevel(LogLevel level) { s_logLevel = level; }
+
+bool Log::enabled(LogLevel level) {
+#ifdef __clang__  // Clang does not appreciate s_doingDebug being the result of
+                  // a macro trick.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunreachable-code"
+#endif
+  return (((s_doingDebug && level == LogLevel::Debug) ||
+           GEODE_HIGHEST_LOG_LEVEL >= level) &&
+          s_logLevel >= level);
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+}
+
+void Log::log(LogLevel level, const char* msg) {
```

Comment:
You need to make darn sure that these are going to get inlined now that they are not in the header or we might start having a real performance issue.

Reply:
We're replacing the logger with a logger library, so this point is moot.

Reply:
Wasn't under the impression this would require the heavier-weight process with a proposal etc.  We're not planning to modify the API or the output of the logger, only replace the code in the Log class, which is _one file_.  If you disagree, let me know and I'll write something up and bring it to the dev list.

Reply:
Introduction of new dependencies should have a discussion. That discussion can be a simple as a PR. In this case I’m asking to about a potential negative impact caused by this change and being told it’s being addressed in a wholesale replacement of logging. How as a community member am I supposed to be aware of this intent. In the mean time of this PR is accepted we don’t know the impact this intermediate change is going to have. No effort has been expended to quantify the cost of this change.

Reply:
Benchmarks are available, inlining of the logger had no runtime performance impact.